moirae/estimators/online/utils/model.py

```diff
@@ -0,0 +1,153 @@
+                 transients: GeneralContainer,
+                 input_template: InputQuantities,
+                 asoh_inputs: Tuple[str]):
+        super().__init__(model, asoh, transients, input_template)
+
+        # Store the information about the identity of variables in the transient state
+        self.asoh_inputs = asoh_inputs
+        self.num_transients = transients.to_numpy().shape[1]
+        self.num_asoh = asoh.get_parameters(self.asoh_inputs).shape[1]
+
+    @property
+    def num_hidden_dimensions(self) -> int:
+        return self.num_transients + self.num_asoh
+
+    def create_hidden_state(self, asoh: HealthVariable, transients: GeneralContainer) -> np.ndarray:
+        """Transform the state of health and transients states (quantities used by CellModel)
+        into the "hidden state" vector used by the actual filter
+
+        Args:
+            asoh: Values of the ASOH parameter
+            transients: Values of the transient states
+        Returns:
+            A hidden state vector ready for use in a filter
+        """
+
+        return np.concatenate([
+            transients.to_numpy(),
+            asoh.get_parameters(self.asoh_inputs)
+        ], axis=1)
+
+    def create_cell_model_inputs(self, hidden_states: np.ndarray) -> Tuple[HealthVariable, GeneralContainer]:
+        """Convert the hidden states into the forms used by CellModel
+
+        Args:
+            hidden_states: Hidden states as used by the estimator
+        Returns:
+            - ASOH with values from the hidden states
+            - Transients state from the hidden states
+        """
+
+        # Update any parameters for the transient state
+        my_transients = self.transients.model_copy(deep=True)
+        batch_transients = np.repeat(self.transients.to_numpy(), axis=0, repeats=hidden_states.shape[0])
+        my_transients.from_numpy(batch_transients)
+
+        # Update the ASOH accordingly
+        my_asoh = self.asoh.model_copy(deep=True)
+        my_asoh.update_parameters(hidden_states[:, self.num_transients:], self.asoh_inputs)
+        return my_asoh, my_transients
+
+    def update_hidden_states(self,
+                             hidden_states: np.ndarray,
+                             previous_controls: np.ndarray,
+                             new_controls: np.ndarray) -> np.ndarray:
+        # Transmute the controls and hidden state into the form required for the CellModel
+        previous_inputs = self.input_template.model_copy(deep=True)
+        previous_inputs.from_numpy(previous_controls)
+        new_inputs = self.input_template.model_copy(deep=True)
+        new_inputs.from_numpy(new_controls)
+
+        my_asoh, my_transients = self.create_cell_model_inputs(hidden_states)
+
+        # Produce an updated estimate for the transient states, hold the ASOH parameters constant
+        output = hidden_states.copy()
+        new_transients = self.model.update_transient_state(previous_inputs, new_inputs=new_inputs,

```

Comment:
Should we pass `previous_inputs` as a keyword argument?

Reply:
That makes sense to me